lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -1976,8 +1975,16 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
           return success();
         }
 
-        rewriter.replaceOpWithNewOp<Torch::AtenSqueezeOp>(binder.op, resultType,
-                                                          gather);
+        // 0 indicesRank will always squeeze the axis dim
+        // Use PrimsSqueezeOp for the case of result with dynamic shape
+        SmallVector<Value> dimList({index});
+        Value dimValueList = rewriter.create<Torch::PrimListConstructOp>(
+            loc,
+            rewriter.getType<Torch::ListType>(
+                rewriter.getType<Torch::IntType>()),
+            dimList);
+        rewriter.replaceOpWithNewOp<Torch::PrimsSqueezeOp>(
+            binder.op, resultType, gather, dimValueList);
```

Comment:
Why use `PrimsSqueezeOp` when the dim list only has one element? You should be able to use the util function `Torch::squeezeTensor(rewriter, binder.op, axis, gather)`, which is what the decomposition for `PrimsSqueezeOp` ends up using anyway.

Reply:
I didn't know the util function. So when the indices is a scalar, it will select 1 from the axis dim and squeeze it. The result rand is `q + (r - 1)` where `q` is input rand and `r` is indices rank. I changed it `AtenSqueezeDimOp` instead.